OMCompiler/SimulationRuntime/c/linearization/linearize.cpp

```diff
@@ -584,18 +584,23 @@ int linearize(DATA* data, threadData_t *threadData)
         free(matrixCz);
         free(matrixDz);
     }
-
-    /* Use the result file name rather than the model name so that the linear file name can be changed with the -r flag, however strip _res.mat from the filename */
-    filename = string(data->modelData->resultFileName) + ".mo";
+    switch(data->modelData->linearizationDumpLanguage){
```

Comment:
What we could do here might be to have a couple of #defines maybe generated earlier (Similar to what @phannebohm  suggested. 

#define OMC_LINEARIZE_DUMP_LANGUAGE_MODELICA 0
#define OMC_LINEARIZE_DUMP_LANGUAGE_MATLAB 1
#define OMC_LINEARIZE_DUMP_LANGUAGE_JULIA 2

A bit verbose but better then 0, 1, 2 and so on

Reply:
done.

Please look at my changes and see if that is what you had in mind!

Reply:
Sweet, it's readable and relatively lightweight.